docs/new-curriculum/js-core-1/week-1/notes.md

```diff
@@ -0,0 +1,349 @@
+
+:::
+
+### Reusing instructions
+
+Let’s consider another scenario. Instead of adding or multiplying numbers, we’ll consider a number like `10.3`.  
+Given the number `10.3`, we could ask:
+
+> 🤔 "What is the nearest whole number to `10.3`?"
+
+The process of finding the nearest whole number to 10.3 is called **rounding**. So we could rephrase the previous question to be:
+
+> 🤔 "What does the number `10.3` **round** to?”
+
+Again we can use Node REPL in order to round the number 10.3 to the nearest whole number. However, in this case, there is no arithmetical operator for rounding the number `10.3` in JavaScript.
+However, we will often want to carry out some kind of operation or series of operations repeatedly. For example, we will often want to round numbers again and again. So in this case we can use a **function**.
+
+:::note
+
+Definition: A function is a reusable set of instructions.
+
+:::
+
+In particular, we can make use of a function called `Math.round`. Recalling the definition, a function is a reusable set of instructions: so `Math.round` contains instructions for rounding any number. Functions will often take **inputs** and then **apply their set of instructions to the inputs** in order to produce an **output**.
+
+So we can write `Math.round` in the Node REPL:
+
+![round-img](./assets/math.round-img.png)
+
+After we've hit enter to evaluate, we then get the following:
+
+![round-img](./assets/math.round-evaluated.png)
+
+So in this case, the REPL output `[Function: round]` is indicating that `Math.round` is a function.
+
+### Calling a function
+
+Once we have the name for a function, we need to get Node to read the function's instructions and execute them. Execution means the computer reads the instructions and carries them out. So we can write the following in the REPL:
+
+![round-img](./assets/math.round-number-10.3.png)
+
+Notice now we have `(` and `)` parentheses after the name of the function and a number inside the parentheses. The parentheses indicate that we are calling the function. The number inside the parentheses is the **input** we're passing to the function.
+
+:::note
+
+Calling a function means telling the computer to read the function's instructions and carry out its instructions. When calling a function we can also pass inputs to the function.
+
+:::
+
+`Math.round(10.3)` is a call expression: it can be thought of as saying apply the set of instructions for `Math.round` to the number `10.3`.
+
+If we type `Math.round(10.3)` then again we get the result 10. So once more we can say that `Math.round(10.3)` evaluates to 10.
+
+:::note
+Important to note that `Math.round` is **implementation opaque**. **Implementation opaque** means we can’t read the set of instructions for `Math.round`. This is because `Math.round` was authored by the developers that created NodeJS.
```

Comment:
I would call it "abstraction" or "encapsulation". It's not so much that we can't go and find out what it does, but also that we don't want to find out (like http - nobody reaaaally wants to know how http works under the hood and fortunately we don't have to)

Reply:
I think we should just cut this paragraph: it feels like something a teacher can cover themselves if it comes up as a question.

There's other things that I would prefer to cover over this.